### PR TITLE
CE-211 separate integration tests into separate job within deploy workflow

### PR DESCRIPTION
**Pull request by benjiwheeler**
### Resolves:

private Jira ticket: https://scratchfoundation.atlassian.net/browse/CE-211

### Changes:

separates integration tests into a separate job within deploy workflow.

as of the initial version of this change, I'm not sure of a few things:
* this code will currently not automatically run integration tests as part of the deploy steps. But we probably want to put those back in, so that integration tests run as part of the deploy. What I'm wondering is, is it better to have separate build/deploy and integration test jobs, which run in sequence? Or is it better to just put the integration tests back into the deploy jobs?
* I'm not totally clear on when to store test results, and when not to. In the deploy steps, we have linting and unit tests; should we attempt to store test results at the end of the deploy steps, as we did previously? Or are they really only useful for integration tests?

### Test Coverage:

n/a, this deals with tests themselves.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -98,8 +107,12 @@ aliases:
 jobs:
   build-and-deploy-staging:
     <<: *build_and_deploy
+  integration-tests-staging:
+    <<: *integration_tests_and_store
   build-and-deploy-production:
     <<: *build_and_deploy
```
Comment (benjiwheeler) [inline .circleci/config.yml:113]: Should we also perform integration tests here? Yes, right?

File: .circleci/config.yml [hunk 2]
```diff
@@ -74,15 +74,24 @@ aliases:
           <<: *setup_python
       - run:
           <<: *deploy
-      - run:
-          <<: *integration
       - store_test_results:
```
Comment (benjiwheeler) [inline .circleci/config.yml:77]: Is it the right decision to continue to store test results here?

File: .circleci/config.yml [hunk 3]
```diff
@@ -98,8 +107,12 @@ aliases:
 jobs:
   build-and-deploy-staging:
     <<: *build_and_deploy
+  integration-tests-staging:
+    <<: *integration_tests_and_store
   build-and-deploy-production:
     <<: *build_and_deploy
+  integration-tests-production:
+    <<: *integration_tests_and_store
```
[the remark this answers is not in the document]
Comment (benjiwheeler) [reply .circleci/config.yml:115]: Good point, I guess you can include the same job under different workflows, and just provide each the correct context and filters? I'll try that

File: .circleci/config.yml [hunk 4]
```diff
@@ -98,8 +107,12 @@ aliases:
 jobs:
   build-and-deploy-staging:
     <<: *build_and_deploy
```
[the remark this answers is not in the document]
Comment (benjiwheeler) [reply .circleci/config.yml:109]: Ah yes, that's right... I was confused when I finally put this up! The integration tests will run as part of those workflows. And then those jobs can be rerun, separately, if we want. (At least that's my hope.)

File: .circleci/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (delasare) [reply .circleci/config.yml]: Yep thats correct you should be able to filter these.

**Review by cwillisf — APPROVED** [review]
Looks good to me! Thanks for doing this!

**Pull request merged**